.github/workflows/validate-csharp.yml

```diff
@@ -0,0 +1,77 @@
+name: CSharp
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+
+env:
+  FORCE_COLOR: 3
+
+jobs:
+  test:
+    name: Test [${{ matrix.os }}]
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v3
+        with:
+          dotnet-version: 6.0.x
+
+      - name: dotnet test Facebook.Yoga.Tests
+        run: dotnet test
+        working-directory: csharp/tests/Facebook.Yoga.Tests
+
+  build:
+    name: Build and Package [${{ matrix.os }}]
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [windows-latest]
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup
+        uses: ./.github/actions/setup-js
+
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v3
+        with:
+          dotnet-version: 6.0.x
+
+      - name: dotnet pack
+        run: dotnet pack -c Release
+        working-directory: csharp
+
+  build-native:
+    name: Build Native Components [${{ matrix.os }}]
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        include:
+          - os: windows-latest
+            rid: win-x64
+          - os: ubuntu-latest
+            rid: linux-x64
```

Comment:
```suggestion
          - os: windows-latest
            rid: win-x64
          - os: windows-latest
            rid: win-arm64
          - os: ubuntu-latest
            rid: linux-x64
          - os: ubuntu-latest
            rid: linux-arm64
          - os: ubuntu-latest
            rid: android-arm64
          - os: macos-latest
            rid: ios-arm64
```

Reply:
We can't do this for all of these platforms until we set up native builds for each platform. Rid here is mainly for target folder. The native build doesn't yet have cross-build infrastructure set up. If we could get the one entry-point that is C#-specific into the regular yoga code-base, then we could chain together the builds and let the regular yoga build handle all of the cross-building for yoga platforms and this step would just package the files into NuGet packages.

Reply:
Ah, got it.